src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/CollectDumpActionTests.cs

```diff
@@ -0,0 +1,127 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Diagnostics.Tools.Monitor.CollectionRules.Options.Actions;
+using System.Threading.Tasks;
+using Xunit;
+using Microsoft.Diagnostics.Tools.Monitor.CollectionRules.Actions;
+using System.Threading;
+using System;
+using System.IO;
+using System.Globalization;
+using Xunit.Abstractions;
+using Microsoft.Diagnostics.Monitoring.TestCommon;
+using Microsoft.Diagnostics.Monitoring.TestCommon.Runners;
+using Microsoft.Diagnostics.Monitoring.TestCommon.Options;
+using Microsoft.Diagnostics.Monitoring.WebApi.Models;
+using static Microsoft.Diagnostics.Monitoring.Tool.UnitTests.EndpointUtilities;
+
+namespace Microsoft.Diagnostics.Monitoring.Tool.UnitTests
+{
+    public sealed class CollectDumpActionTests
+    {
+        private const string TempEgressDirectory = "/tmp";
+        private const string ExpectedEgressProvider = "TmpEgressProvider";
+
+        private ITestOutputHelper _outputHelper;
+        private readonly EndpointUtilities _endpointUtilities;
+
+        public CollectDumpActionTests(ITestOutputHelper outputHelper)
+        {
+            _outputHelper = outputHelper;
+            _endpointUtilities = new(_outputHelper);
+        }
+
+        [Theory]
+        [InlineData(DumpType.Full)]
+        [InlineData(DumpType.Mini)]
+        [InlineData(DumpType.Triage)]
+        [InlineData(DumpType.WithHeap)]
+        [InlineData(null)]
+        public async Task CollectDumpAction_Success(DumpType? dumpType)
+        {
+            DumpType ExpectedDumpType = (dumpType != null) ? dumpType.Value : CollectDumpOptionsDefaults.Type;
```

Comment:
If we use a collection rule (as suggested in another comment), then we use `AddCollectDumpAction` which takes a `DumpType` and an egress provider. It looks like the extension method uses `DumpType` instead of `DumpType?`, though I see no reason why it couldn't be changed to accept an optional value. Since `DumpType` cannot be `null`, do we want to change the extension method to take a `DumpType?` to avoid needing to do `DumpType ExpectedDumpType = dumpType.GetValueOrDefault(CollectDumpOptionsDefaults.Type);`?

Reply:
Yeah, change the extension method parameter to be `DumpType? type = null` and move it to the end. We'll want to keep the `egress` parameter without a default value since it is required.